lib/users/api.js

```diff
@@ -637,11 +631,15 @@ class API {
   async downloadJobOutput(jobId, outputPath = undefined) {
     if (!outputPath) {
       let job = await this.getJobDetails(jobId);
-      outputPath = job.filename;
+      outputPath = job.output_filename;
     }
     let stream = fs.createWriteStream(outputPath);
-    stream.on('error', function(error) { throw error; });
-    stream.on('end', function() { return; });
+    stream.on('error', function(error) {
+      throw error;
+    });
+    stream.on('end', function() {
+      return;
```

Comment:
Are these necessary at all since they're not doing anything?

Reply:
some of this was written by David, who never really grasped how exception handling works in Node. Definitely possible these could be removed

Reply:
I'm specifically concerned about the `end` handler, which I don't think is needed. You are correct that the `error` handler will probably not propagate the exception correctly, though.